src/hooks/utility.rs

```diff
@@ -92,7 +102,46 @@ pub async fn process_utility_hook(
 }
 
 fn is_support_utility(stmt_type: NodeTag) -> bool {
-    stmt_type == pg_sys::NodeTag::T_ExplainStmt
+    stmt_type == pg_sys::NodeTag::T_ExplainStmt || stmt_type == pg_sys::NodeTag::T_ViewStmt
+}
+
+fn view_query(query_string: &core::ffi::CStr) -> Result<bool> {
+    // Get the current schema in Postgres
+    let current_schema = get_postgres_current_schema();
+    // Set DuckDB search path according search path in Postgres
+    set_search_path_by_pg()?;
+
+    let dialect = PostgreSqlDialect {};
+    let statements = Parser::parse_sql(&dialect, query_string.to_str()?)?;
+    // visit statements, capturing relations (table names)
+    let mut visited = vec![];
+
+    visit_relations(&statements, |relation| {
+        visited.push(relation.clone());
+        ControlFlow::<()>::Continue(())
+    });
+
+    for relation in visited.iter() {
+        let (schema_name, relation_name) = if relation.0.len() == 1 {
+            (current_schema.clone(), relation.0[0].to_string())
+        } else if relation.0.len() == 2 {
+            (relation.0[0].to_string(), relation.0[1].to_string())
+        } else {
+            error!(
+                "it is not possible to create a view with more than 2 parts in the relation name"
+            );
+        };
+
+        if !view_exists(&relation_name, &schema_name)? {
+            fallback_warning!(format!(
+                "{schema_name}.{relation_name} does not exist in DuckDB"
```

Comment:
What is most likely to cause this error? Trying to create a VIEW on a table that doesn't exist? Could you please add a comment?

Reply:
If attempting to create a view on a table that does not exist in DuckDB, do not push it down to DuckDB.